### PR TITLE
[DTMF Dialler] Enable DTMF mode and dialing out mode in dialpad

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Remove code that forces user's keyboard to end of phone number
# Why

allows contoso to edit the phone number with the devices keyboard

https://skype.visualstudio.com/SPOOL/_workitems/edit/3553803
# How Tested

Locally in sample and storybook

### CODE REVIEW

**Comment by JamesBurnside** [comment]
Adding @carocao-msft that introduced this to see if there's a reason to keep it

**Review by carocao-msft — APPROVED** [review]

**Review by carocao-msft — COMMENTED** [review]
Code looks good, but better double check with Alex to make sure this doesn't affect PSTN.
Design decision was made before to make sure cursor always stays at end of textfield
https://skype.visualstudio.com/SPOOL/_workitems/edit/3015832

File: packages/react-components/src/components/Dialpad/Dialpad.tsx [hunk 1]
```diff
@@ -341,19 +374,20 @@ const DialpadContainer = (props: {
     >
       <TextField
         styles={concatStyleSets(textFieldStyles(theme), props.styles?.textField)}
-        value={textFieldValue ? textFieldValue : formatPhoneNumber(plainTextValue)}
+        componentRef={inputBoxReference}
```
Comment (mgamis-msft) [inline packages/react-components/src/components/Dialpad/Dialpad.tsx:377]: Just curious what this reference is used for?

Comment (dmceachernmsft) [reply packages/react-components/src/components/Dialpad/Dialpad.tsx:377]: ah leftover from something I was trying I will remove it

**Review by mgamis-msft — APPROVED** [review]

**Pull request merged**